src/Logary.Facade/Facade.fs

```diff
@@ -226,27 +226,194 @@ type LoggingConfig =
+        else
+            let nextIndex = nextInvalidCharIndex + 1
+            let propInsidesRng = Range(first + 1, nextIndex - 2)
+            match tryGetPropInRange template propInsidesRng with
+            | prop when not (obj.ReferenceEquals(prop, Property.Empty)) -> foundProp prop
+            | _ -> foundText (Range.Substring(template, first, (nextIndex - 1)))
+            nextIndex
+
+  let parseParts (template:string) foundText foundProp =
+      let tlen = template.Length
+      let rec go start =
+          if start >= tlen then ()
+          else match Internal.findNextNonPropText start template foundText with
+                  | next when next <> start -> go next
+                  | _ -> go (Internal.findPropOrText start template foundText foundProp)
+      go 0
+
 module internal Formatting =
   open System.Text
   open System.Text.RegularExpressions
 
   [<Literal>]
   let FieldExnKey = "exn"
 
-  let fieldRegex =
-    // https://msdn.microsoft.com/en-us/library/bs2twtah.aspx#balancing_group_definition
-    Regex(@"( (?<B>\{) [\s\w]+ (?<-B>\}) )", RegexOptions.IgnorePatternWhitespace ||| RegexOptions.Compiled)
-
-  let extractMatches fields (template : string) =
-    fieldRegex.Matches template
-    |> Seq.cast<Match>
-    |> Seq.toList
-    |> List.choose (fun m ->
-      let name = m.Value.Trim([| '{'; '}'; ' '|])
-      if fields |> Map.containsKey name then
-        Some (name, fields |> Map.find name, m)
-      else
-        None)
+  let literateFormatValue (context: LiterateContext) (fields: Map<string,obj>) = function
+    | Event template ->
+      let themedParts = ResizeArray<string * LiterateToken>()
+      let matchedFields = ResizeArray<string>()
+      let foundText (text: string) = themedParts.Add (text, Text)
+      let foundProp (prop: FsMtParser.Property) =
+        match Map.tryFind prop.Name fields with
+        | Some propValue ->
+          // render using string.Format, so the formatting is applied
+          let stringFormatTemplate = prop.AppendPropertyString(StringBuilder(), "0").ToString()
+          let fieldAsText = System.String.Format (stringFormatTemplate, [| propValue |])
+          // find the right theme colour based on data type
+          let valueColour =
+            match propValue with
+            | :? bool -> KeywordSymbol
+            | :? int16 | :? int32 | :? int64 | :? decimal | :? float -> NumericSymbol
+            | :? string | :? char -> StringSymbol
+            | _ -> OtherSymbol
+          if context.PrintTemplateFieldNames then
+            themedParts.Add ("["+prop.Name+"] ", Subtext)
+          matchedFields.Add prop.Name
+          themedParts.Add (fieldAsText, valueColour)
+
+        | None ->
+          // if the property wasn't provided, just render the missing value
+          // but using 'LevelFatal' so it really stands out.
+          themedParts.Add (prop.ToString(), LevelFatal)
+
+      FsMtParser.parseParts template foundText foundProp
```

Comment:
I don't understand this code without code-completion - does `FsMtParser.parseParts template foundText foundProp` return a function that you give a set to? Or are you mutating something?